capsules/src/usb/cdc.rs

```diff
@@ -0,0 +1,477 @@
+                },
+            ],
+        ];
+
+        let (device_descriptor_buffer, other_descriptor_buffer) =
+            descriptors::create_descriptor_buffers(
+                descriptors::DeviceDescriptor {
+                    vendor_id: VENDOR_ID,
+                    product_id: PRODUCT_ID,
+                    manufacturer_string: 1,
+                    product_string: 2,
+                    serial_number_string: 3,
+                    class: 0x2, // Class: CDC
+                    max_packet_size_ep0: max_ctrl_packet_size,
+                    ..descriptors::DeviceDescriptor::default()
+                },
+                descriptors::ConfigurationDescriptor {
+                    ..descriptors::ConfigurationDescriptor::default()
+                },
+                interfaces,
+                endpoints,
+                None, // No HID descriptor
+                Some(cdc_descriptors),
+            );
+
+        Cdc {
+            client_ctrl: ClientCtrl::new(
+                controller,
+                device_descriptor_buffer,
+                other_descriptor_buffer,
+                None, // No HID descriptor
+                None, // No report descriptor
+                LANGUAGES,
+                STRINGS,
+            ),
+            buffers: [
+                Buffer64::default(),
+                Buffer64::default(),
+                Buffer64::default(),
+            ],
+            tx_buffer: TakeCell::empty(),
+            tx_len: Cell::new(0),
+            tx_remaining: Cell::new(0),
+            tx_offset: Cell::new(0),
+            tx_client: OptionalCell::empty(),
+            rx_buffer: TakeCell::empty(),
+            rx_len: Cell::new(0),
+            rx_offset: Cell::new(0),
+            rx_client: OptionalCell::empty(),
+        }
+    }
+
+    #[inline]
+    fn controller(&self) -> &'a C {
+        self.client_ctrl.controller()
+    }
+
+    #[inline]
+    fn buffer(&'a self, i: usize) -> &'a [VolatileCell<u8>; 64] {
+        &self.buffers[i - 1].buf
+    }
+}
+
+impl<'a, C: hil::usb::UsbController<'a>> hil::usb::Client<'a> for Cdc<'a, C> {
+    fn enable(&'a self) {
+        // Set up the default control endpoint
+        self.client_ctrl.enable();
+
+        // Setup buffers for IN and OUT data transfer.
+        self.controller()
+            .endpoint_set_in_buffer(ENDPOINT_IN_NUM, self.buffer(ENDPOINT_IN_NUM));
+        self.controller()
+            .endpoint_in_enable(TransferType::Bulk, ENDPOINT_IN_NUM);
+
+        self.controller()
+            .endpoint_set_out_buffer(ENDPOINT_OUT_NUM, self.buffer(ENDPOINT_OUT_NUM));
+        self.controller()
+            .endpoint_out_enable(TransferType::Bulk, ENDPOINT_OUT_NUM);
+    }
+
+    fn attach(&'a self) {
+        self.client_ctrl.attach();
+    }
+
+    fn bus_reset(&'a self) {
+        // Should the client initiate reconfiguration here?
+        // For now, the hardware layer does it.
+
+        debug!("Bus reset");
```

Comment:
Should we remove this? For new board bring-up debug prints can cause panics and are a pain.